rayon-core/src/scope/mod.rs

```diff
@@ -608,9 +621,7 @@ impl<'scope> ScopeBase<'scope> {
 
     /// Executes `func` as a job, either aborting or executing as
     /// appropriate.
-    ///
-    /// Unsafe because it must be executed on a worker thread.
-    unsafe fn complete<FUNC, R>(&self, owner: Option<&WorkerThread>, func: FUNC) -> R
+    fn complete<FUNC, R>(&self, owner: Option<&WorkerThread>, func: FUNC) -> R
```

Comment:
Why is this not unsafe? I guess it never had to be...?

Reply:
Well, at least the comment "because it must be executed on a worker thread" is optional after #844. I didn't really change that functionality in this PR, just cleaning up.

I suppose there's a correctness issue if `Some(owner)` doesn't match the latch, but we do have `debug_assert_eq` for that, and it's not really a *safety* concern. Worst case is that we can't wake up the right thread. We control all the callers in this same module though, and they do trivially match from `new` to `complete`.

(We can't put the `&WorkerThread` directly in the scope or latch because it's not `Sync`.)